src/fileseq/filesequence.py

```diff
@@ -150,13 +154,17 @@ def padding(self):
     def setPadding(self, padding):
         """
         Set new padding characters for the sequence.
-        i.e. "#" or "@@@", or an empty string to disable range formatting.
+        i.e. "#" or "@@@" or '%04d', or an empty string to disable range formatting.
 
         :type padding: str
         :rtype: None
         """
         self._pad = padding
-        self._zfill = sum([PAD_MAP[c] for c in self._pad])
+        printf_len = PRINTF_RE.split(self._pad, 1)
+        if len(printf_len) == 3:
+            self._zfill = int(printf_len[1])
+        else:
+            self._zfill = sum([PAD_MAP[c] for c in self._pad])
```

Comment:
I would leave `getPaddingChars` as-is, since our primary padding format is "#" and "@", whereas the printf-style is more of an additional input parsing support in order to get it into a `FileSequence` object. Do you know of a need to have padding converted to printf-style for string output?